node.go

```diff
@@ -71,6 +71,7 @@ type Node struct {
 	leftNode      *Node
 	rightNode     *Node
 	subtreeHeight int8
+	isLegacy      bool
```

Comment:
```suggestion
	isV0Node      bool
```
or similar. legacy is ambiguous?

Reply:
we are using the `legacy` term from design stage, just worried if there are many things have to change